### DISCUSSION TITLE
clock delay of AD5781

### DISCUSSION THREAD

**Discussion by FeiyangLou** [vnegnev/marcos_extras#12] [Debugging]
Sometimes we can't operate the gradient outputs using OCRA1, so we checked the inputs of AD5781
![1684722818277](https://github.com/vnegnev/marcos_extras/assets/103250574/beb284c1-3a9d-49ba-9c2f-b3e166ed1a8d)
But after changing the update rate of gradient DAC to a smaller value it goes good

**Comment by vnegnev** [comment]
Sorry about the delay. I'm a bit confused:
- What are the time units in the two oscilloscope photos?
- Is the problem that the clock is seemingly misaligned with the SYNC in the first screenshot (the SYNC goes low after a clock cycle has passed), or something else? 
- In the second screenshot, I don't understand what you mean by 'before and after the RC filter' - why is there a time shift of ~1 cycle but no shift in amplitude or shape of the waveform? 
- What does the data line look like for some test data
- At which update rate do you see a change in behaviour? Is a similar change visible if you run your experiment with the marga simulation model, or only on physical hardware?

**Reply by FeiyangLou** [reply]
- time unit is 1μs/div
- Yes, they should be aligned
- It's a test that showed clk signal delayed by one cycle after the RC filter, while sync didn't. 
- 
![image](https://github.com/vnegnev/marcos_extras/assets/103250574/97cac3e2-eabe-4524-85b0-24a8f8c77daf)
Sorry I forgot to post the data picture.
These pictures were taken on our physical hardware under 0.2 grad_max_update_rate, which was default in Experiments.py. After changing the update rate to 0.01 the signals became correct, then we changed to 0.1, finally back to 0.2 and all is good. We haven't tried the marga simulation. 
As OCRA1 works well now we decided not to change anything until next time it doesn't operate. So maybe you can just ignore now, thx! 😄
